controller.py

```diff
@@ -297,15 +298,14 @@ def email(self):
                 "The contact URI for this library is missing or invalid"
             )
 
-        try:
-            hyperlink.notify(self.emailer, self.app.url_for)
-        except SMTPException, e:
-            # We were unable to send the email.
-            return INTEGRATION_ERROR.detailed(
-                _("SMTP error while sending email to %(address)s",
-                  address=hyperlink.resource.href)
-            )
-        return Response(unicode(library.internal_urn), 200)
+        validation, is_new = get_one_or_create(self._db, Validation, resource=hyperlink.resource)
+        validation.restart()
+        validation.mark_as_successful()
+
+        if (validation.success):
+            return Response(unicode(library.internal_urn), 200)
+        else:
+            return CANNOT_VALIDATE
```

Comment:
Looking at the code I don't see how validation.success would ever be false here, so the code could be removed without hurting anything. Plus, leaving it in would mean testing it, which would be annoying.